plugins/woocommerce/includes/wccom-site/rest-api/class-wc-rest-wccom-site-installer-error-codes.php

```diff
@@ -0,0 +1,90 @@
+<?php
+/**
+ * WCCOM Site Installer Error Codes Class
+ *
+ * @package WooCommerce\WCCom\API
+ * @since   7.7.0
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * WCCOM Site Installer Error Codes Class
+ *
+ * Stores data for errors, returned by installer API.
+ */
+class WC_REST_WCCOM_Site_Installer_Error_Codes {
+
+	const NOT_AUTHENTICATED                   = 'not_authenticated';
+	const NO_ACCESS_TOKEN                     = 'no_access_token';
+	const NO_SIGNATURE                        = 'no_signature';
+	const SITE_NOT_CONNECTED                  = 'site_not_connnected';
+	const INVALID_TOKEN                       = 'invalid_token';
+	const REQUEST_VERIFICATION_FAILED         = 'request_verification_failed';
+	const USER_NOT_FOUND                      = 'user_not_found';
+	const NO_PERMISSION                       = 'forbidden';
+	const IDEMPOTENCY_KEY_MISMATCH            = 'idempotency_key_mismatch';
+	const NO_INITIATED_INSTALLATION_FOUND     = 'no_initiated_installation_found';
+	const ALL_INSTALLATION_STEPS_RUN          = 'all_installation_steps_run';
+	const REQUESTED_STEP_ALREADY_RUN          = 'requested_step_already_run';
+	const PLUGIN_ALREADY_INSTALLED            = 'plugin_already_installed';
+	const INSTALLATION_ALREADY_RUNNING        = 'installation_already_running';
+	const INSTALLATION_FAILED                 = 'installation_failed';
+	const FILESYSTEM_REQUIREMENTS_NOT_MET     = 'filesystem_requirements_not_met';
+	const FAILED_GETTING_PRODUCT_INFO         = 'product_info_failed';
+	const INVALID_PRODUCT_INFO_RESPONSE       = 'invalid_product_info_response';
+	const WCCOM_PRODUCT_MISSING_SUBSCRIPTION  = 'wccom_product_missing_subscription';
+	const WCCOM_PRODUCT_MISSING_PACKAGE       = 'wccom_product_missing_package';
+	const WPORG_PRODUCT_MISSING_DOWNLOAD_LINK = 'wporg_product_missing_download_link';
```

Comment:
@denho I just noticed that this `WPORG_PRODUCT_MISSING_DOWNLOAD_LINK` code is missing in the `ERROR_MESSAGES` array below. Is that intentional?

Reply:
Not intentional, but that's fine. It was originally designed that messages and HTTP codes are optional, and there is a fallback to default value:
https://github.com/woocommerce/woocommerce/blob/56b63ada60e209003fb69524c76c7e8630392b0a/plugins/woocommerce/includes/wccom-site/rest-api/class-wc-rest-wccom-site-installer-error.php#L25-L26